src/wasm-linker.h

```diff
@@ -132,14 +90,16 @@ class Linker {
     assert(implementedFunctions.count(name));
   }
 
-  // Allocate the user stack, set up the initial memory size of the module, lay
-  // out the linear memory, process the relocations, and set up the indirect
-  // function table.
-  void layout();
+  bool isEmpty() {
```

Comment:
i don't like the ad-hoc nature of this function. how about a boolean starting false that is flipped to true when we do any processing, and return that boolean here?